src/main/java/org/ibp/api/java/impl/middleware/inventory/manager/TransactionServiceImpl.java

```diff
@@ -156,6 +160,9 @@ public void updatePendingTransactions(final List<TransactionUpdateRequestDto> tr
 	public void saveDeposits(final LotDepositRequestDto lotDepositRequestDto, final TransactionStatus transactionStatus) {
 		final WorkbenchUser user = this.securityService.getCurrentlyLoggedInUser();
 
+		if (lotDepositRequestDto.getSourceStudy() != null) {
+			this.studyValidator.validate(lotDepositRequestDto.getSourceStudy(), false);
```

Comment:
Since we're making experiment transaction record for the study, I think study should be unlocked if study is specified.

Reply:
Done